addons/web_editor/static/src/js/wysiwyg/widgets/image_crop_widget.js

```diff
@@ -1,174 +1,238 @@
+    saveAttributes: [
+        'resModel',
+        'resId',
+        'attachmentId',
+        'originalId',
+        'originalName',
+        'mimetype',
+    ],
 
     /**
      * @constructor
      */
-    init: function (parent, options, media) {
-        var self = this;
+    init(parent, options, media) {
+        this._super(...arguments);
         this.media = media;
-        this.$media = $(this.media);
-        var src = this.$media.attr('src').split('?')[0];
-        this.aspectRatioList = [
-            [_t("Free"), '0/0', 0],
-            ["16:9", '16/9', 16 / 9],
-            ["4:3", '4/3', 4 / 3],
-            ["1:1", '1/1', 1],
-            ["2:3", '2/3', 2 / 3],
+        this.$media = $(media);
+        // Needed for editors in iframes.
+        this.document = media.ownerDocument;
+        // Used for res_model and res_id
+        this.options = options;
+        this.aspectRatios = [
+            {label: _t("Free"), value: 0},
+            {label: "16:9", value: 16 / 9},
+            {label: "4:3", value: 4 / 3},
+            {label: "1:1", value: 1},
+            {label: "2:3", value: 2 / 3},
         ];
-        this.imageData = {
-            imageSrc: src,
-            originalSrc: this.$media.data('crop:originalSrc') || src, // the original src for cropped DB images will be fetched later
-            mimetype: this.$media.data('crop:mimetype') || (_.str.endsWith(src, '.png') ? 'image/png' : 'image/jpeg'), // the mimetype for DB images will be fetched later
-            aspectRatio: this.$media.data('aspectRatio') || this.aspectRatioList[0][1],
-            isExternalImage: src.substr(0, 5) !== 'data:' && src[0] !== '/' && src.indexOf(window.location.host) < 0,
-        };
-        this.options = _.extend({
-            title: _t("Crop Image"),
-            buttons: this.imageData.isExternalImage ? [{
-                text: _t("Close"),
-                close: true,
-            }] : [{
-                text: _t("Save"),
-                classes: 'btn-primary',
-                click: this.save,
-            }, {
-                text: _t("Discard"),
-                close: true,
-            }],
-        }, options || {});
-        this._super(parent, this.options);
-        this.trigger_up('getRecordInfo', _.extend(this.options, {
-            callback: function (recordInfo) {
-                _.defaults(self.options, recordInfo);
-            },
-        }));
+        const src = this.media.getAttribute('src');
+        const data = Object.assign({}, media.dataset);
+        // Needed to restore src on discard, src of the image before the crop.
+        this.initialSrc = src;
+        // This part of the crop state is not inside cropperData so we need to maintain it manually
+        this.aspectRatio = parseFloat(data.aspectRatio) || 0;
+        this.mimetype = data.mimetype || (_.str.endsWith(src, '.png') ? 'image/png' : 'image/jpeg');
```

Comment:
Note: you can use `endsWith` now, you are touching the line, you become responsible :D